expr/expr.go

```diff
@@ -1090,32 +1095,42 @@ func EvalExpr(e *expr, from, until int32, values map[MetricRequest][]*MetricData
 			}
 
 			denominator = denominators[0]
-		} else if len(numerators) == 2 && len(e.args) == 1 {
-			numerator = numerators[0]
-			denominator = numerators[1]
+		} else if len(firstArg) == 2 && len(e.args) == 1 {
+			numerators = append(numerators, firstArg[0])
+			denominator = firstArg[1]
 		} else {
 			return nil, errors.New("must be called with 2 series or a wildcard that matches exactly 2 series")
 		}
 
-		if numerator.StepTime != denominator.StepTime || len(numerator.Values) != len(denominator.Values) {
-			return nil, errors.New("series must have the same length")
+		for _, numerator := range numerators {
+			if numerator.StepTime != denominator.StepTime || len(numerator.Values) != len(denominator.Values) {
+				return nil, errors.New("series must have the same length")
+			}
 		}
 
-		r := *numerator
-		r.Name = fmt.Sprintf("divideSeries(%s)", e.argString)
-		r.Values = make([]float64, len(numerator.Values))
-		r.IsAbsent = make([]bool, len(numerator.Values))
+		for _, numerator := range numerators {
+			r := *numerator
+			if useMetricNames {
+				r.Name = fmt.Sprintf("divideSeries(%s,%s)", numerator.Name, denominator.Name)
+			} else {
+				r.Name = fmt.Sprintf("divideSeries(%s)", e.argString)
```

Comment:
I would prefer the new naming in all cases and have `target=divideSeries(metric.[ab]*)` become named eg. `divideSeries(metric.alice,metric.bob)` rather than remain `divideSeries(metric.[ab]*)` particularly in the case of a bare `*`. Am I in the minority?